### PR TITLE
Create a pipeline for MRTK3 API reference docs generation

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview
Create a new pipeline to generate the binaries for API reference docs.

### CODE REVIEW

**Comment by Zee2** [comment]
You're our damn savior @MaxWang-MS, you're a hero

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

**Comment by MaxWang-MS** [comment]
I can remove the pr run and just leave the ci run after people approve this PR:)

File: Pipelines/pr.yaml [hunk 1]
```diff
@@ -29,3 +34,16 @@ jobs:
     parameters:
       BuildStandalone: true
       BuildUWP: true
+
+- job: DocsBinariesGeneration
+  pool: Unity_2021.3.3f1_Pool
+  steps:
+  - checkout: self
+
+  - checkout: PipelineTools
```
Comment (keveleigh) [inline Pipelines/pr.yaml:43]: What's used in this job from the `PipelineTools` repo?

File: Pipelines/ci.yaml [hunk 2]
```diff
@@ -17,6 +17,11 @@ resources:
     endpoint: ToolsAccess
     name: tools.internal
     ref: 78ecf591964ecf511b4c7a0cd84d98ce11e8be16
+  - repository: DocToolUnityProject
+    type: git
+    endpoint: ToolsAccess
+    name: lib.doctools
+    ref: refs/heads/main
```
Comment (keveleigh) [inline Pipelines/ci.yaml:24]: Do we want to lock this into a specific commit, so we aren't potentially inadvertently broken by a change to the repo? That way, any change made in the lib.doctools repo will need to be explicitly adopted here, which feels a bit more stable.

File: Pipelines/pr.yaml (hunk 1 above)
Comment (MaxWang-MS) [reply Pipelines/pr.yaml:43]: Almost everything related to docs gen minus the yaml files (as you can see I am only checking in yaml files in this PR). We are exporting a unity package from that repo during CI time (using a script in that repo), and then import the exported package into the MRTK3 Unity project where we run the command to generate the binaries (also using a script in that repo).

File: Pipelines/ci.yaml (hunk 2 above)
Comment (MaxWang-MS) [reply Pipelines/ci.yaml:24]: I thought about that, but I sort of want to use MRTK3 as the "CI" for that repo to catch inadvertent changes to that repo.

File: Pipelines/pr.yaml (hunk 1 above)
Comment (keveleigh) [reply Pipelines/pr.yaml:43]: I might be getting this backwards, but I thought the docs gen stuff was all in the DocToolUnityProject repo

File: Pipelines/ci.yaml (hunk 2 above)
Comment (keveleigh) [reply Pipelines/ci.yaml:24]: I'd be happy to work with you to set up a real CI for that repo 😀 I'd prefer not to potentially break MRTK's pipelines based on someone making an inadvertent change to a different repo.

File: Pipelines/pr.yaml (hunk 1 above)
Comment (MaxWang-MS) [reply Pipelines/pr.yaml:43]: Oh yes you are right, the docs gen stuff was in the lib.doctools repo. However, we still need the pipeline tools repo to set up Unity license correctly (please refer to line 47 below). Without that I ran into Unity licensing issues.

File: Pipelines/ci.yaml (hunk 2 above)
Comment (MaxWang-MS) [reply Pipelines/ci.yaml]: Sounds great! I have updated the yaml files to point to a specific commit.

**Review by keveleigh — APPROVED** [review]

**Pull request merged**